be/src/runtime/routine_load/data_consumer.cpp

```diff
@@ -344,6 +344,10 @@ Status KafkaDataConsumer::get_partition_meta(std::vector<int32_t>* partition_ids
     RdKafka::Metadata* metadata = nullptr;
     RdKafka::ErrorCode err = _k_consumer->metadata(false /* all_topics */, topic, &metadata, timeout);
     if (err != RdKafka::ERR_NO_ERROR) {
+        if (_k_event_cb.get_error_msg().empty()) {
+            // some authentication errors event can only be triggered when the consumer is closed.
+            _k_consumer->close();
```

Comment:
shall check `closed()` before invoking `close()` ?

Reply:
no need to check. 
if an error is reported here, it will be closed here(new added for error events) or later.
new requests will use a new consumer.

Reply:
a little confused between this change and the change in ~KafkaDataConsumer. Is it any possibility the consumer get closed from other places. And whether the `close()` call is idempotent. If idempotent, the check of closed() in ~KafkaDataConsumer is useless. If not, the code here implies that no one will close it before under any circumstance.

Reply:
`close` is used to get some error messages here.
i add some closed DCHECK in other places where the consumer is used.